jbpm-wb-process-runtime/jbpm-wb-process-runtime-backend/src/main/java/org/jbpm/workbench/pr/backend/server/RemoteProcessRuntimeDataServiceImpl.java

```diff
@@ -98,52 +96,22 @@ public ProcessInstanceSummary getProcessInstance(String serverTemplateId,
-        List<RuntimeLogSummary> logs = new ArrayList<RuntimeLogSummary>(processInstanceHistory.size() + allTaskEventsByProcessInstanceId.size());
-        PrettyTime prettyDateFormatter = new PrettyTime();
-
-        for (int i = processInstanceHistory.size() - 1; i >= 0; i--) {
-            NodeInstance nis = processInstanceHistory.get(i);
-
-            if (nis.getNodeType().equals("HumanTaskNode")) {
-                logs.add(new RuntimeLogSummary(nis.getId(),
-                                               prettyDateFormatter.format(nis.getDate()),
-                                               "Task '" + nis.getName() + "' was created",
-                                               "System"));
-                for (TaskEventInstance te : allTaskEventsByProcessInstanceId) {
-                    if (te.getWorkItemId() != null && te.getWorkItemId().equals(nis.getId()) &&
-                            (te.getType().equals("CLAIMED") || te.getType().equals("RELEASED") || te.getType().equals("COMPLETED"))) {
-                        logs.add(new RuntimeLogSummary(nis.getId(),
-                                                       "- " + prettyDateFormatter.format(te.getLogTime()),
-                                                       "Task '" + nis.getName() +
-                                                               "' was " + te.getType().toLowerCase() + " by user " + te.getUserId(),
-                                                       "Human"));
-                    }
-                }
-            } else if (nis.getNodeType().equals("StartNode")) {
-                logs.add(new RuntimeLogSummary(nis.getId(),
-                                               prettyDateFormatter.format(nis.getDate()),
-                                               "Process '" + processName + "' was created",
-                                               "Human"));
-            } else if (nis.getNodeType().equals("EndNode")) {
-                logs.add(new RuntimeLogSummary(nis.getId(),
-                                               prettyDateFormatter.format(nis.getDate()),
-                                               "Process '" + processName + "' was completed",
-                                               "System"));
-            }
-        }
-        return logs;
+        ProcessServicesClient processServicesClient = getClient(serverTemplateId,
+                                                                ProcessServicesClient.class);
+
+        List<NodeInstance> processInstanceHistory = processServicesClient.findNodeInstances(deploymentId,
+                                                                                            processInstanceId,
+                                                                                            0,
+                                                                                            100);
```

Comment:
@mswiderski we got a few places like in here where we need the complete list of items to be retrieved from a paginated service response. Just having a limited/fixed number of items to retrieve isn't a good solution. I wonder if we could define something like -1 for page size to bypass pagination and return all items to avoid multiple calls or that is something you would really prefer to avoid?